### DISCUSSION TITLE
:fireworks: DOC is the new G.O.A.T. :fireworks:

### DISCUSSION THREAD

**Discussion by pmatran** [pymartheproject/pymarthe#23] [Show and tell]
Hi everyone :wave:,
-------------------------------------------------------------------------------------------------------------------
<br>

:speaking_head: _“Software quality is measured by the weight of its documentation.”_
-------------------------------------------------------------------------------------------------------------------

You're probably aware that the quality of a librairy, particularly one in the style of _pymarthe_, is closely linked to the value of the **documentation** provided by its developers. **PyMarthe v1.0** currently has a significant number of functions and tools.
Until now, only the `Guide_v1.0.py` file gave _users_/_modellers_ an overview of the classic manipulations that could be performed with the library. It is in _script_ format, unillustrated and poorly organised. It is **ABSOLUTLY NECESSARY** to create documentation that is **clear**, **organised**, _**user-friendly**_ and accessible in 1 click :stuck_out_tongue_winking_eye:!

:loud_sound: That the reason why, a new **branch** has just been created: [`feature/Update-guides`](https://github.com/apryet/adeqwat/tree/feature/Update-Guides). Its main purpose is to build a **organized**,  _**user-friendly**_  documentation as  `jupyter notebook`.

For the moment, **2 documentation guides** have been added:

&emsp;&emsp;⫸&nbsp;_`Basic_usage.ipynb`_ : ultimate guide to learn how to deal with **MARTHE** files, properties, grid geometry, field, outputs...<br>
&emsp;&emsp;⫸&nbsp;_`Advance_usage.ipynb`_ : ultimate guide to learn how to properly interface **MARTHE** and **PEST** algorithms.<br>

<br>

To get more familiar with the guides,  just start one and follow the instructions over each section:
```shell
jupyter notebook examples/Basic_usage.ipynb
```

:warning: **WARNINGS** :warning:
Some additional fixes **MUST BE INTEGRATED** before playing around with new guides. Make sure to have a look on recent #22 .

<br>

:bulb: Perspectives & ideas
-------------------------------------------------------------------------------------------------------------------

The librairy functionalities have more or less already been explained and documented, I really think the goal now is to provided **end-to-end** & **reproducable** examples (from scratch). It would be wonderfull to add the modified _hallue_ parametrisation example provided by @apryet at the last conference/workshop (_ENSEGID, 2022_), don't you think?

Once more feel free to comment, criticize, fix, test and improve those new guides before merging.

Thank's for all,

@pmatran

**Comment by apryet** [comment]
Hello @pmatran,

Thank you for your time and relevant recommendations, I fully concur: bug fixes & docs should be priority #1.

The thing is, we are overwhelmed by a bunch of other things and for the moment, the only contribution I can decently do is to fix bugs as they come to me. 

That being said, it is clear that the library should be consolidated before being expanded with new functionalities. 

I have planned a full clean-up for this fall, with an official release of a distribution through PyPi.

Thank you again !

**Reply by pmatran** [reply]
Thank's for your reply @apryet :+1:,

I totally agree and understand, time is becoming more and more precious, this is the main drawback of every **open source projects**. Personally, I will probably stay in touch, feel free to contact me in private for any support or help especially  for the `release` part.

Take care!
